R/GenerateRefTables.R

```diff
@@ -303,3 +303,118 @@ UpdateDetLimitRef <- function() {
+           "Quality Control Sample-Lab Re-Analysis",
+           "Quality Control Sample-Lab Split",
+           "Quality Control-Meter Lab Duplicate",
+           "Quality Control-Meter Lab Duplicate 2",
+           "Sample-Routine Resample")
+  blank <- c("Quality Control Field Sample Equipment Rinsate Blank",
+             "Quality Control Lab Sample Equipment Rinsate Blank",
+             "Quality Control Sample-Equipment Blank",
+             "Quality Control Sample-Field Ambient Conditions Blank",
+             "Quality Control Sample-Field Blank",
+             "Quality Control Sample-Lab Blank",
+             "Quality Control Sample-Post-preservative Blank",
+             "Quality Control Sample-Pre-preservative Blank",
+             "Quality Control Sample-Reagent Blank",
+             "Quality Control Sample-Trip Blank",
+             "Quality Control-Meter Lab Blank",
+             "Quality Control-Negative Control",
+             "Sample-Depletion Replicate",
+             "Sample-Negative Control")
+  cal <- c("Quality Control Field Calibration Check",
+           "Quality Control Field Msr/Obs Post-Calibration",
+           "Quality Control Field Msr/Obs Pre-Calibration",
+           "Quality Control Sample-Field Spike",
+           "Quality Control Sample-Field Surrogate Spike",
+           "Quality Control Sample-Lab Continuing Calibration Verification",
+           "Quality Control Sample-Lab Control Sample/Blank Spike",
+           "Quality Control Sample-Lab Control Sample/Blank Spike Duplicate",
+           "Quality Control Sample-Lab Control Standard",
+           "Quality Control Sample-Lab Control Standard Duplicate",
+           "Quality Control Sample-Lab Initial Calib Certified Reference Material",
+           "Quality Control Sample-Lab Initial Calibration Verification",
+           "Quality Control Sample-Lab Matrix Spike",
+           "Quality Control Sample-Lab Matrix Spike Duplicate",
+           "Quality Control Sample-Lab Spike",
+           "Quality Control Sample-Lab Spike Duplicate",
+           "Quality Control Sample-Lab Spike Target",
+           "Quality Control Sample-Lab Spike of a Lab Blank",
+           "Quality Control Sample-Lab Surrogate Control Standard",
+           "Quality Control Sample-Lab Surrogate Control Standard Duplicate",
+           "Quality Control Sample-Lab Surrogate Method Blank",
+           "Quality Control Sample-Measurement Precision Sample",
+           "Quality Control Sample-Reference Sample",
+           "Quality Control-Calibration Check",
+           "Quality Control-Calibration Check Buffer",
+           "Sample-Positive Control")
+  other <- c("Quality Control Sample-Other")
+  
+  WQXActivityTypeRef <- raw.data%>%
+    dplyr::mutate(TADA.ActivityType.Flag = dplyr::case_when(
+      Code %in% dup ~ "QC_duplicate",
+      Code %in% blank ~ "QC_blank",
+      Code %in% cal ~ "QC_calibration",
+      Code %in% other ~ "QC_other"
```

Comment:
To align with the other flag functions, suggest giving non-QC samples a Flag as well. Either "Not_QC" or similar.

Reply:
good call! - I thought about doing that but some other flag functions leave the flag column blank for valid samples - is this something we should change for all flag functions? or just the ones that use a reference table?